HighlightUtilities/functions.php

```diff
@@ -222,11 +222,25 @@ function splitCodeIntoArray($html)
         throw new \UnexpectedValueException("The given HTML could not be parsed correctly.");
     }
 
-    $xpath = new \DOMXPath($dom);
-    $spans = $xpath->query("//span[contains(text(), '\n') or contains(text(), '\r\n')]");
+    $spans = $dom->getElementsByTagName('span');
 
     /** @var \DOMElement $span */
     foreach ($spans as $span) {
+        if ($span->hasChildNodes()) {
+            $hasNewlines = false;
+
+            foreach ($span->childNodes as $child) {
+                if ($child->nodeType === XML_TEXT_NODE && preg_match('/\R/', $child->textContent)) {
+                    $hasNewlines = true;
+                    break;
+                }
+            }
+
+            if (!$hasNewlines) {
+                continue;
+            }
+        }
```

Comment:
Wouldn't this be the same as:

```suggestion
        if ($span->hasChildNodes() && ! preg_match('/\R/', $span->textContent)) {
            continue;
        }
```

Since `DOMElement::$textContent` includes the concatenation of all descendant text nodes.

Reply:
In the original attempt, this was not quite the case, by using `DOMElement::$textContent` it would include child `<span>`s. I wanted to find child nodes that were purely text and not additional spans.

But I wrote a different implementation dropping the DOM parser entirely.